src/libraries/System.Private.CoreLib/src/System/MemoryExtensions.Trim.Utf8.cs

```diff
@@ -0,0 +1,77 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace System
+{
+    public static partial class MemoryExtensions
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static ReadOnlySpan<byte> TrimUtf8(this ReadOnlySpan<byte> span)
+        {
+            // Assume that in most cases input doesn't need trimming
```

Comment:
The aggressive inlining and this comment appear to come from the UTF16 implementation, where the checks for whether there's anything to trim are minimal.  For UTF8, it doesn't look so minimal.

Reply:
Removed.